src/sysen2/mldev.108

```diff
@@ -94,69 +96,95 @@ GO:	MOVE P,[-LPDLL-1,,PDL-1]
 	move c,[sixbit /ITSNMS/]
 	.getsys b,
 	 .value
-	;; is it DIRxx? See if xx is an ITS name, then make it xxDIR instead
-	hlrz c,a
+	;; check for 4-character devices. Only one checked for is SHOE, which only works with 2-character
+	;;  hostnames.
+	ldb c,[003000,,a]		; get last four characters
+	camn c,[sixbit /  SHOE/]	; is it the SHOE device?
+	  jrst doshoe
+
+	;; devices that are DIRhh or DIRhhh are converted to hhDIR or hhhDIR for later consistent handling.
+	hlrz c,a		; a is for example, 'DIREX'; c becomes '   DIR'
 	cain c,'DIR
-	 jrst [ hrlz b,a	;get xx part in b
-		pushj p,isitsp
-		 jrst .+1
-		;; b is an ITS, make device xxDIR
-		lsh c,-12.
-		dpb b,[.bp (777700),c]
-		movem c,jbcdev
-		move a,b	; get host in A for gothst
-		jrst gothst ]
-	;; is it Xxx? allow this as alias for "Xperimental" xx
+	  jrst dodir
+
+	;; is it Xhhx or Xhhh? allow this as alias for "Xperimental" hh or hhh
 	ldb c,[.bp (770000),a]
 	cain c,'X
 	 jrst [	move b,a
 		lsh b,6		;get rid of X
-		pushj p,isitsp
-		 jrst .+1
-		movem b,jbcdev
-		move a,b	;get host in A for gothst
-		jrst gothst ]
-	;; is it xxDIR?
+		pushj p,isitsp	; see if we now have a known ITS name
+		 jrst nox	;  nope, skip handling as experimental case
+		move c,dskdev	; assume DSK device
+		movem c,jbcdev	; store as remote device
+		move a,b	; get host in A for gothst
+		jrst gothst ]	; and go process request for this host
+
+nox:				; jump here if hostname minus first "x" is not a known ITS
+				; check to see if format is hhDIR
 	move c,a
-	lsh c,12.
-	cain c,'DIR
-	 jrst [ ldb b,[.bp (777700),a] ;get xx part in b
-		pushj p,isitsp
-		 jrst .+1
-		move a,b	; get host in A
-		jrst gothst ]		
-	move b,a
-	;; is b an ITS name?
-	pushj p,isitsp
-	 jrst notits
-	  jrst gothst
+	lsh c,12.		; move past hostname
```

Comment:
This only handles HHDIR, not HHHDIR. Why is DIR treated specially, shouldn't all devices (including DIR) be treated uniformly?

Reply:
You must be looking at the old code.  Please look again at the latest.  That code fragment no longer appears in MLDEV 108.

Reply:
At this point, if C was 'EXDIR ', the LSH makes it 'DIR', the JRST DIRCHK continues the check for HHDIR, since we know that we have 'xxDIR ' at this point.  If we skip, however, which would be the case with, HHHDIR, we continue on with the following code:
```
        lsh c,12.               ; move past hostname
        camn c,dirdev           ; is what remains the DIR device?
          jrst dirchk           ;  yes, continue checking hhDIR case

                                ; is it hhhdir or hhh
        hrlz c,a                ; get device in c
        jumpe c, host3          ; jump if no device
     ```
     
     This part distinguishes between HHHDIR and HHH cases (we know we don't have HHDIR at this point). 
     
    But you are right that DIR is being handled specially here, but the HHDIR and HHSHO (for example) cases should be treated the same -- e.g. as HHDDD.  The reason DIR is handled specially is because in the original code it was as well.  But you are probably right that I could just check for HHDDD and HHHDDD where DDD = 0, and jump to code that would handle any device, including DIR.